### DISCUSSION TITLE
Getting kafka metrics

### DISCUSSION THREAD

**Discussion by YasiruR** [IBM/sarama#2622] [Q&A]
0

I've been trying to get kafka broker metrics using go sarama library and I have few concerns.

I initialized a sarama config using sarama.NewConfig() with the version as sarama.V2_0_0_0 and I used this config to obtain metrics as mentioned in your [doc](https://godoc.org/github.com/Shopify/sarama)

`metrics.GetOrRegisterMeter("incoming-byte-rate", config.MetricRegistry).RateMean()`

1. Does this provide the total incoming byte rate including all the incoming messages? Because I compared this value with lenses and other tools but it is totally different (same for "outgoing-byte-rate" and "request-rate". I suppose this request rate implies the message rate).

2. Unlike `RateMean()` function, `Rate1()`, `Rate5()` and `Rate15()` return only 0 though there's plenty of activity in the kafka cluster. Do you know any reason for this?

3. I used `metrics.GetOrRegisterMeter(fmt.Sprintf("incoming-byte-rate-for-broker-%v", broker.ID()), config.MetricRegistry).RateMean()` to get broker level metrics as mentioned in the [doc](https://godoc.org/github.com/Shopify/sarama) but I only received 0 (same for other broker level metrics as well).

I'd really appreciate if someone can help me out here and enlighten me on these concerns. Thanks in advance. :)

**Comment by ghost** [comment]
Thank you for taking the time to raise this issue. However, it has not had any activity on it in the past 90 days and will be closed in 30 days if no updates occur.
Please check if the master branch has already resolved the issue since it was raised. If you believe the issue is still valid and you would like input from the maintainers then please comment to ask for it to be reviewed.
